tools/c7n_azure/c7n_azure/actions/logic_app.py

```diff
@@ -0,0 +1,90 @@
+# Copyright 2019 Microsoft Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from azure.mgmt.logic import logic_management_client
+
+from c7n.actions.webhook import Webhook
+from c7n.utils import type_schema
+
+"""Calls an Azure Logic App with optional parameters
+   and body populated from JMESPath queries.
+
+    .. code-block:: yaml
+
+      policies:
+        - name: call-logic-app
+          resource: azure.vm
+          description: |
+            Call logic app with list of VM's
+          actions:
+           - type: logic-app
+             resource-group: custodian-test
+             logic-app-name: cclogicapp
+             batch: true
+             body: 'resources[].{ vm_name: name }'
+"""
+
+
+class LogicAppAction(Webhook):
+
+    schema = type_schema(
+        'logic-app',
+        required=['resource-group', 'logic-app-name'],
+        **{
+            'resource-group': {'type': 'string'},
+            'logic-app-name': {'type': 'string'},
+            'body': {'type': 'string'},
+            'batch': {'type': 'boolean'},
+            'batch-size': {'type': 'number'},
+            'query-params': {
```

Comment:
do you need query-params  & headers?

Reply:
Yeah I was initially going to remove but then I found that you can access both of them in the logic app and in some cases it seems to be a useful way to send values, so I left them there.